### PR TITLE
fix: gofmt replace interface with any

### PR DESCRIPTION
**Pull request by notanatol**
### Checklist

- [X] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)

### Description
In the spirit of go1.18 the old `interface{}` becomes `any`.

It also looks like there's a change in syntax of the build tags, I think it's good to keep up with these changes.

commands ran:

```
go fix ./pkg/... ./cmd/... 
gofmt -r 'interface{} -> any' -w .
```

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3103)

### CODE REVIEW

File: cmd/bee/cmd/start_windows.go [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build windows
```
Comment (acud) [inline cmd/bee/cmd/start_windows.go:5]: is this needed?

**Review by acud — APPROVED** [review]
LGTM minus the build flag change?

File: cmd/bee/cmd/start_windows.go (hunk 1 above)
Comment (notanatol) [reply cmd/bee/cmd/start_windows.go:5]: It seems that starting go1.18 the syntax for build constraints has changed https://pkg.go.dev/cmd/go#hdr-Build_constraints. It's likely that the old syntax will be phased out in the future so we need to get on-board at some point, might as well be now. @vandot  wdyt?

Comment (notanatol) [reply cmd/bee/cmd/start_windows.go:5]:
```
In Go 1.18, go fix [go tool fix] now removes the now-obsolete // +build lines in modules declaring go 1.18 or later in their go.mod files.
```

**Review by istae — APPROVED** [review]

File: cmd/bee/cmd/start_windows.go (hunk 1 above)
Comment (vandot) [reply cmd/bee/cmd/start_windows.go:5]: As @notanatol said, it started in go1.17 and probably they will deprecate old syntax at some point

**Review by acud — APPROVED** [review]

**Pull request merged**